hazelcast/src/test/java/com/hazelcast/config/XMLConfigPreProcessorTest.java

```diff
@@ -500,6 +501,56 @@ public void testXmlDeniesDuplicateMemberAttributesConfig() throws Exception {
         configBuilder.build();
     }
 
+    @Test
+    public void testXmlVariableReplacementAsSubstring() throws Exception {
+        Properties properties = new Properties();
+        String xml = "<hazelcast>\n" +
+                "    <properties>\n" +
+                "        <property name=\"${env}-with-suffix\">local-with-suffix</property>\n" +
+                "        <property name=\"with-prefix-${env}\">with-prefix-local</property>\n" +
+                "    </properties>\n" +
+                "</hazelcast>";
+
+        properties.setProperty("env", "local");
+
+        ByteArrayInputStream bis = new ByteArrayInputStream(xml.getBytes());
+        XmlConfigBuilder configBuilder = new XmlConfigBuilder(bis);
+        configBuilder.setProperties(properties);
+        Config config = configBuilder.build();
+
+        assertEquals(config.getProperty("local-with-suffix"), "local-with-suffix");
+        assertEquals(config.getProperty("with-prefix-local"), "with-prefix-local");
+    }
+
+    @Test
+    public void testXmlImportWithVariableReplacementAsSubstring() throws Exception {
+        Properties properties = new Properties();
```

Comment:
Thanks.

Please keep in mind: readability for tests is extremely important because they are an executable specification if your system. If they are more difficult to read than needs to be, the loose their value because people have a hard time figuring out the purpose of a test.

So for normal code readability is important, but for tests imho even more so.